main.md

```diff
@@ -861,6 +968,37 @@ large `jti` values or store only a hash thereof.
 Note: To accommodate for clock offsets, the server MAY accept DPoP
 proofs that carry an `iat` time in the reasonably near future (e.g., a few
 seconds in the future).
+Because clock skews between servers and clients may be large,
+servers may choose to limit DPoP proof lifetimes by using
+server-provided nonce values rather than clock times,
+yielding intended results even in the face of arbitrarily large clock skews.
+
+Server-provided nonces are an effective means of preventing DPoP proof replay.
+
+## DPoP Proof Pre-Generation (#Pre-Generation)
+
+An attacker in control of the client can pre-generate DPoP proofs for use
+arbitrarily far into the future by choosing the `iat` value in the
+DPoP proof to be signed by the proof-of-possession key.
+Note that one such attacker is the person who is the legitimate user of the client.
+The user may pre-generate DPoP proofs to exfiltrate
+from the machine possessing the proof-of-possession key
+upon which they were generated
+and copy them to another machine that does not possess it.
+For instance, a bank employee might pre-generate DPoP proofs
+on a bank computer and then copy them to another machine
+for use in the future, thereby bypassing bank audit controls.
+When DPoP proofs can be pre-generated and exfiltrated,
+all that is actually being proved in DPoP protocol interactions
+is proof of possesion of a DPoP proof -- not of the proof-of-possession key.
+
+Use of server-provided nonces can prevent this attack.
+By providing new nonce values at times of its choosing,
```

Comment:
... nonce values that are not predictable by an attacker ...

Reply:
Done - thanks